src/redirect.rs

```diff
@@ -0,0 +1,108 @@
+//! HTTP redirection endpoint.
+//!
+//! # Examples
+//!
+//! ```no_run
+//! # use futures::executor::block_on;
+//! # fn main() -> Result<(), std::io::Error> { block_on(async {
+//! #
+//! use tide::Redirect;
+//!
+//! let mut app = tide::new();
+//! app.at("/").get(|_| async move { Ok("meow") });
+//! app.at("/nori").get(Redirect::temporary("/"));
+//! app.listen("127.0.0.1:8080").await?;
+//! #
+//! # Ok(()) }) }
+//! ```
+
+use crate::utils::BoxFuture;
+use crate::StatusCode;
+use crate::{Endpoint, Request, Response};
+
+/// A redirection endpoint.
+///
+/// # Example
+///
+/// ```
+/// # use tide::{Response, Redirect, Request, StatusCode};
+/// # fn next_product() -> Option<String> { None }
+/// # #[allow(dead_code)]
+/// async fn route_handler(request: Request<()>) -> tide::Result {
+///     if let Some(product_url) = next_product() {
+///         Ok(Redirect::new(product_url).into())
+///     } else {
+///         //...
+/// #       Ok(Response::new(StatusCode::Ok)) //...
+///     }
+/// }
+/// ```
+#[derive(Debug, Clone)]
+pub struct Redirect<T: AsRef<str>> {
+    status: StatusCode,
+    location: T,
+}
+
+impl<T: AsRef<str>> Redirect<T> {
+    /// Creates an endpoint that represents a redirect to `location`.
+    ///
+    /// Uses status code 302 Found.
+    pub fn new(location: T) -> Self {
+        Self {
+            status: StatusCode::SeeOther,
+            location,
```

Comment:
This can probably be improved by having `location` `.parse()` into a `HeaderName` already?

Reply:
@Fishrock123 
Not sure what you're getting at. The location argument on this method does not become a `HeaderName`.

Can you clarify?

Reply:
Whoops, my bad. I think I was probably thinking `HeaderValue` but `set_header` does that under the hood anyway.